yesod-core/src/Yesod/Core/Types.hs

```diff
@@ -306,6 +306,20 @@ data Content = ContentBuilder !BB.Builder !(Maybe Int) -- ^ The content and opti
              | ContentFile !FilePath !(Maybe FilePart)
              | ContentDontEvaluate !Content
 
+contentToTruncatedString :: Content -> Int -> String
+contentToTruncatedString (ContentBuilder builder maybeLength) maxLength =
+    let
+      truncated = take maxLength (show builder)
```

Comment:
hmm. `show` is probably not what we want here, but it may be. Calling `show` on a string-ish type usually will add quotes around it, so it can be copy/pasted into Haskell code.

```
λ> import Data.ByteString.Builder
λ> show ("hello" :: Builder)
"\"hello\""
λ> print ("hello" :: Builder)
"hello"
λ> putStrLn "hello"
hello
λ> print ("hello" :: String)
"hello"
```

(`print` is `putStrLn . show`)

Likewise, `show "\n" == "\"\\n\""`

But, what we have is a bunch of bytes. Do we interpret them as ASCII and do `Data.ByteString.Char8.unpack :: ByteString -> String`? Or do we treat them as UTF8 and do `Data.Text.Encoding.decodeUtf8` ?

Reply:
If we accepted `TypedContent` here instead of `Content`, we should be able to infer the encoding. If it's ASCII or UTF-8 or some other text encoding, the right thing to do will be obvious. And if it's _not_ text, maybe we shouldn't include the snippet at all.

Reply:
Great call